include/clink/kernels/opdefs/types.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2021 The Clink Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CLINK_FEATURE_OPDEFS_TYPES_H_
+#define CLINK_FEATURE_OPDEFS_TYPES_H_
+
+#include "mlir/IR/Types.h"
+
+namespace clink {
+
+class ModelType
+    : public mlir::Type::TypeBase<ModelType, mlir::Type, mlir::TypeStorage> {
+public:
+  using Base::Base;
+};
+
+class VectorType
```

Comment:
Maybe replace VectorType with SparseVector as we discussed. Please feel free to check how TFRT handles the DenseHostTensor in MLIR.

Reply:
`DenseHostTensor` is registered as `TensorType` in TFRT, and there seems not to be a `DenseHostTensorType` or `HostTensorType`. Maybe it is because no other subclasses of `Tensor` is used as parameter or return value of TFRT kernels. 

With that said, I agree that we should change our usage here to `SparseVectorType` as it is guaranteed that there will be both `SparseVector` and `DenseVector`. I'll make the change.

Reply:
It looks like `coo.convert_coo_to_dht` is a kernel that takes a `CooHostTensor` as input and returns a `DenseHostTensor` as output. Could you check if we could follow its practice to support `DenseVector` and `SparseVector`?